### PR TITLE
Display not 100% available OS as disabled in dropdown menu

### PR DESCRIPTION
**Pull request by vuntz**
Depends on https://github.com/crowbar/barclamp-provisioner/pull/375

### CODE REVIEW

File: crowbar_framework/app/models/node_object.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def self.default_platform
   end
 
   def self.available_platforms
-    @@available_platforms ||= begin
+    @available_platforms ||= begin
```
Comment (tboerger) [inline crowbar_framework/app/models/node_object.rb]: Are you sure that you have to remove the second ```@```? I think that is only the case for ```self << class```

Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: I tried, and yes, it was needed.

File: crowbar_framework/app/helpers/form_helper.rb [hunk 2]
```diff
@@ -19,7 +19,8 @@ module FormHelper
   def platforms_for_select(selected)
     options_for_select(
       available_platforms.map { |p| [crowbar_service.pretty_target_platform(p), p] },
-      selected.to_s
+      selected: selected.to_s,
+      disabled: disabled_platforms.map { |p| p }
```
Comment (tboerger) [inline crowbar_framework/app/helpers/form_helper.rb]: Why the map?

Comment (vuntz) [inline crowbar_framework/app/helpers/form_helper.rb]: Just some over-aggressive copy&paste :-) Fixed

**Comment by dirkmueller** [comment]
@vuntz looks like tests need to be adjusted, could you check?

File: crowbar_framework/app/models/node_object.rb [hunk 3]
```diff
@@ -485,6 +498,10 @@ def allocate!
     return if @node.nil?
     return if @role.nil?
     return if self.allocated?
+
+    raise "Platform #{self.target_platform} is not fully setup for installation yet." if NodeObject.disabled_platforms.include? self.target_platform
```
Comment (tboerger) [inline crowbar_framework/app/models/node_object.rb]: Maybe swap this? Unavailable platforms are more important :)

**Comment by vuntz** [comment]
@dirkmueller I'm removing the commit that fails the tests for now. Will resubmit (and fix tests) in a later PR so we can go ahead and merge this if we want it now.

**Comment by vuntz** [comment]
Anyone up for a review?

**Comment by tboerger** [comment]
:+1:

**Pull request merged**